.travis.yml

```diff
@@ -12,22 +12,28 @@ matrix:
       addons: &gcc49
         apt:
           sources: ['ubuntu-toolchain-r-test']
-          packages: ['g++', 'gcc', 'gfortran']
+          packages: ['g++', 'gcc', 'gfortran', 'valgrind']
       env:
         - CXX='g++'
         - CC='gcc'
         - FC='gfortran'
         - TRAVIS_TEST=1  # this is a workaround to trigger include in src/CMakeLists.txt
+        - CTEST_OUTPUT_ON_FAILURE=1
 
 install:
   - pip install -r requirements.txt
 
 script:
-  - ./setup
+  - ./setup 
   - cd build
   - make -j2
   - make install
-  - ctest -j2
+  - ctest -j2 -E valgrind 
+  - ./setup --type=debug build-debug
```

Comment:
I think we should not run both release and debug inside one but rather run a matrix of two build.

Reply:
So you mean that both the debug build and the release build sould run all tests?

Reply:
Not necessarily, but they should not run inside the same test but be two independent tests from the Travis point of view. If this is not clear, then don't worry about it - I can always split this later.